pkg/image/api/helper.go

```diff
@@ -130,14 +130,9 @@ func IsRegistryDockerHub(registry string) bool {
 	}
 }
 
-// ParseDockerImageReference parses a Docker pull spec string into a
-// DockerImageReference.
-func ParseDockerImageReference(spec string) (DockerImageReference, error) {
+func parseDockerImageReference(repoParts []string, spec string) (DockerImageReference, error) {
```

Comment:
Having two different functions with the same name looks suspicious or at least redundant.